symbolic/src/Data/Macaw/Symbolic.hs

```diff
@@ -437,7 +437,7 @@ mkParsedBlockRegCFG archFns halloc posFn b = crucGenArchConstraints archFns $ do
     -- Create atom for entry
     inputAtom <- mmExecST $ CR.mkInputAtoms ng entryPos (Empty :> regType) >>= \case
       Empty :> atm -> return atm
-      _ -> error "Invalid input atom creation for mkParsedBlockRegCFG"
+      _ -> error "Invalid input atom creation for mkParsedBlockRegCFG" -- TODO: fix warning
```

Comment:
```
src/Data/Macaw/Symbolic.hs:440:7: error: [-Woverlapping-patterns, -Werror=overlapping-patterns]
    Pattern match is redundant
    In a case alternative: _ -> ...
    |
440 |       _ -> error "Invalid input atom creation for mkParsedBlockRegCFG"
    |       ^
```

Reply:
arg, on these the coverage checker got smarter, so we need these cases for older ghc but can't have them for newer. I guess it is back to CPP

Reply:
I discussed this issue with @kquick and he indicated that the `panic` package should be used here instead of `error`. Would that negate the need to also use a CPP version check or would that still be required?

Reply:
I think that is independent - GHC is now able to tell that the fallthrough case is impossible (whereas prior versions could not), so we either need or cannot have that case depending on GHC version

Reply:
Right, the use of `panic` instead of `error` should be a different changeset and won't affect the above.  I resolved this via https://github.com/GaloisInc/macaw/pull/172, which you can merge to this branch.

Reply:
The patch for this is to insert some lines such as:
```
#if !MIN_VERSION_base(4,14,0)
   _ -> error "blah blah"
#endif
```
The file may also need a pragma `{-#  LANGUAGE CPP #-}` at the top.
The use of `panic` as opposed to `error` is more of a style thing from my point of view.  I think it's a positive change, but independent.